### PR TITLE
JUnit test AddLifeInsuranceCommandTest and utility class LifeInsuranceBuilder

### PR DESCRIPTION
**Pull request by OscarWang114**
Changes:
- Every field of  `LifeInsurance` is now a unique class of itself. Please refer to the last section.
- Refactor of constructor of `LifeInsurance`
- Name field of `InsurancePerson` is now a `Name` instead of a `String`
- Created JUnit test `AddLifeInsuranceCommandTest` and utility class `LifeInsuranceBuilder`

New type of fields in `LifeInsurance` :
- `InsuranceName` for the name of insurance, regex only accepts alphanumeric letters and cannot be blank.
- `Premium` for premium, regex only accepts numbers and one optional decimal point.
- `ContractFileName` for the name of the contract file, accepts alphanumeric letters, hyphens, underscores, and cannot be blank.

@RSJunior37  Sorry I have to replace your `InsuranceBuilder` with my `LifeInsuranceBuilder` due to the updates in LifeInsurance class. I also updated your other tests to fit with the new model and will not claim credit for them.

@Juxarius  I removed your `LifeInsuranceTest.java`. Really sorry about that but I think you can restore it easily just by changing the parts contructing `LifeInsurance`!

### CODE REVIEW

**Comment by OscarWang114** [comment]
81 percent!!!!!!

File: data/addressbook.xml [hunk 1]
```diff
@@ -150,7 +183,7 @@
                 <insured>Roy Balakrishnan</insured>
                 <beneficiary>Mary Jane</beneficiary>
                 <premium>500.0</premium>
-                <contractName>NotInLisa-Expensive.pdf</contractName>
+                <contractFileName>NotInLisa-Expensive.pdf</contractFileName>
```
Comment (Juxarius) [inline data/addressbook.xml:186]: lol i knew this change was coming

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 2]
```diff
@@ -64,13 +64,13 @@ public AddCommand parse(String args) throws ParseException {
...
+            final Gender gender = addPersonOptionalFieldDescriptor.getGender();
             //@@author
             //@@author OscarWang114
-            Phone phone = addPersonOptionalFieldDescriptor.getPhone();
-            Email email = addPersonOptionalFieldDescriptor.getEmail();
-            Address address = addPersonOptionalFieldDescriptor.getAddress();
+            final Phone phone = addPersonOptionalFieldDescriptor.getPhone();
+            final Email email = addPersonOptionalFieldDescriptor.getEmail();
+            final Address address = addPersonOptionalFieldDescriptor.getAddress();
             ReadOnlyPerson person = new Person(name, phone, email, address, dob, gender, tagList);
```
Comment (Juxarius) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java]: why are we putting as final again? defensive programming?

File: src/main/java/seedu/address/model/insurance/LifeInsurance.java [hunk 3]
```diff
@@ -29,105 +27,98 @@
...
-        this.premiumString = new SimpleStringProperty(this.getPremiumString());
     }
 
     /**
      * Creates a copy of the given ReadOnlyInsurance.
      */
     public LifeInsurance(ReadOnlyInsurance source) {
+        this(source.getInsuranceName(), source.getOwner(), source.getInsured(), source.getBeneficiary(),
+                source.getPremium(), source.getContractFileName(), source.getSigningDate(), source.getExpiryDate());
         this.id = new SimpleObjectProperty<>(source.idProperty().get());
```
Comment (Juxarius) [inline src/main/java/seedu/address/model/insurance/LifeInsurance.java:100]: haha didnt think of this method, good job

File: src/main/java/seedu/address/model/insurance/LifeInsurance.java [hunk 4]
```diff
@@ -207,57 +208,46 @@ public String getBeneficiaryName() {
...
-    public String getContractName() {
-        return contractName.get();
+    public LocalDate getSigningDate() {
+        return signingDate;
     }
 
-    public void setSigningDateString(String signingDateString) throws IllegalValueException {
-        this.signingDate = new DateParser().parse(signingDateString);
-        this.signingDateString.set(requireNonNull(signingDateString));
+    public void setSigningDateString(String signingDateString) {
```
Comment (Juxarius) [inline src/main/java/seedu/address/model/insurance/LifeInsurance.java]: consider making this private?

File: src/main/java/seedu/address/model/insurance/InsurancePerson.java [hunk 5]
```diff
@@ -2,45 +2,49 @@
...
  */
 public class InsurancePerson {
 
-    //TODO: Change from String to Name
-    private StringProperty name;
+    private ObjectProperty<Name> name;
     private Optional<ReadOnlyPerson> person;
 
-    public InsurancePerson(String name) {
+    public InsurancePerson(String nameString) throws IllegalValueException {
```
Comment (Juxarius) [inline src/main/java/seedu/address/model/insurance/InsurancePerson.java:20]: Just wanted to say "Told ya~"

**Review by Juxarius — CHANGES_REQUESTED** [review]
LGTM

only change u need is to change the li tag in UserGuide.adoc to addli
cos currently its wrong and you are tagging nothing
asciidoctor also keeps giving warning for this

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java (hunk 2 above)
Comment (OscarWang114) [reply src/main/java/seedu/address/logic/parser/AddCommandParser.java]: That is what I initially thought. But now I don't think it makes much meaning to use `final` here.

File: src/main/java/seedu/address/model/insurance/LifeInsurance.java (hunk 4 above)
Comment (OscarWang114) [reply src/main/java/seedu/address/model/insurance/LifeInsurance.java]: Yes will do

**Review by Juxarius — APPROVED** [review]

**Pull request merged**